### PR TITLE
implemented logging out feature and associated tests run and passing

### PR DESCRIPTION
**Pull request by clovellbsc**
## Before you start
1. Is there an issue on the repository for this request? If not, please create one.
2. Have you followed the correct workflow in the [team documentation](https://docs.google.com/document/d/13yi709ax4nmfTG9J0F57yEFpYRH7fdtczwvVJCrNIsM/edit#)?
3. Conflict resolution to be dealt with **on GitHub**.

## What do these code changes do?
Logging out feature implemented, which deletes the current session. Edited the delete route to be a post.

## How do these code changes effect the application?
Ability to log out

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have created a PR ticket on the [Trello board](https://trello.com/b/P8lEsvZe/acebook).
- [x] I will link the PR to the Trello card once created.
- [x] I will send the PR in the Slack channel and ping @here.

### CODE REVIEW

**Comment by clovellbsc** [comment]
[I would like to log out of my account](https://trello.com/c/SKETnEnK/16-i-would-like-to-log-out-of-my-account)

File: cypress/integration/user_can_log_out_spec.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  it("A user that has logged out cannot view posts", () => {
+    cy.signUp(firstName, lastName, email, password);
+    cy.login(email, password);
+    cy.get("#dropdownUser1").click();
+    cy.get("#signOut").click();
+    cy.visit("/posts");
+
+    cy.url().should("not.include", "/posts");
+    cy.url().should("include", "/sessions");
+  });
```
Comment (adamwoodcock98) [inline cypress/integration/user_can_log_out_spec.js:26]: Great inclusion of this edge case Chris! Fantastic job.

**Review by adamwoodcock98 — CHANGES_REQUESTED** [review]
Perfect, nice work. Thanks for modernising the seed projects `delete` request. Good coverage of the edge cases too.

The only thing is the brief description of the PR, we need the documentation, but other than that you got this done in no time.

File: cypress/integration/user_can_log_out_spec.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  it("A user that logs out is redirected to /sessions", () => {
+    cy.signUp(firstName, lastName, email, password);
+    cy.login(email, password);
+    cy.get("#dropdownUser1").click();
+    cy.get("#signOut").click();
+
+    cy.url().should("not.include", "/posts");
+    cy.url().should("include", "/sessions");
+  });
```
Comment (PKilgarriff) [inline cypress/integration/user_can_log_out_spec.js:15]: Should we be redirecting to sessions or to sessions/new?

**Review by PKilgarriff — APPROVED** [review]
Having said I wouldn't I have also then messed around with the navbar, so my apologies - once this is merged I'll sort the conflict on the other PR.

Happy to approve this, but just a thought that we should be redirecting to sessions/new not just to sessions

**Review by adamwoodcock98 — APPROVED** [review]
Looks great, I'll leave the conflicts for you or Paul to go through.

**Comment by PKilgarriff** [comment]
Have merged the conflicting changes -  build is passing so I'm merging to main and then will make sure nothing else has secretly broken

**Pull request merged**